src/domains/dashboard/hooks/useAlarmFromUrl.js

```diff
@@ -0,0 +1,35 @@
+import { useMount } from "react-use"
+
+import { useDispatch } from "store/redux-separate-context"
+import { getHashParams } from "utils/hash-utils"
+import { setAlarmAction, setGlobalPanAndZoomAction } from "domains/global/actions"
+
+export default () => {
+  const dispatch = useDispatch()
+  useMount(() => {
+    const params = getHashParams()
+    const alarmWhen = params["alarm_when"]
+    if (alarmWhen) {
+      const alarmTime = Number(alarmWhen)
+
+      const alarmStatus = params["alarm_status"]
+      const alarmChart = params["alarm_chart"]
+      const alarmValue = params["alarm_value"]
+
+      dispatch(setAlarmAction({
+        alarm: {
+          chart: alarmChart,
```

Comment:
Since this is an chartId and not an entire chart, could we rename it to `chartId`?